docs/smart-contracts/guards.md

```diff
@@ -0,0 +1,385 @@
+  (require-capability (ESCROW))
+  (create-account (pact-id) (create-pact-guard "escrow"))
+  (transfer from (pact-id) amount))
+
+(defun complete-escrow (to amount)
+  (require-capability (ESCROW))
+  (with-capability (USER_GUARD (pact-id)) ;; enforces guard on account (pact-id)
+    (transfer (pact-id) to amount)))
+```
+
+## Union guard
+
+```json
+Union = Keyset | KeysetRef | UserGuard | CapabilityGuard | ModuleGuard | PactGuard
+```
+
+## Pact value literals
+
+The following pseudo code provides type information for decoding Pact values returned by guards:
+
+```json
+Union PvLiteral = PvString | PvInteger | PvDecimal | PvBool
+
+PvString = String
+
+# You can treat an integer as a JSON number if it's between the range
+# -9007199254740991 <= n && n <= 9007199254740991 
+# or as a string you need to decode an integer that exceeds the range:
+
+PvInteger = {"int": (Number | String) }
+
+# You can treat an integer as a JSON number if it's between the range
+# -9007199254740991 <= n && n <= 9007199254740991 
+# or as a string you need to decode an integer that exceeds the range:
+
+  The exact logic for how this happens is determined by the function, in pseudo:
+    def encodeInteger (i:PactInteger) : {"int": (Number | String) } =
+      let encodeInt(n:PactInteger) : Number | String = 
+            if -9007199254740991 <= n && n <= 9007199254740991 then encodeNumber else encodeString
+      in return { "int": encodeInt(i) }
+
+PvDecimal = Number | { "decimal":String }
+
+# You can treat a decimal as a JSON number if it's between the range
+# -9007199254740991 <= mantissa(i) && mantissa(i) <= 9007199254740991 
+# or as a string you need to decode an integer that exceeds the range:
+
+  The exact logic for how this happens is determined by the function, in pseudo:
+    def encodeDecimal (i:PactDecimal) : (Number | {"decimal": String }) =
+      if -9007199254740991 <= mantissa(i) && mantissa(i) <= 9007199254740991 then 
+        return encodeNumber(i) 
+       else return { "decimal":encodeString(i) }
+
+PvBool = Bool
+```
+
+## Pact value union types
+
+The following pseudo code provides type information for decoding Pact values returned by recursive guards:
+
+```json
+Union PactValue = PvLiteral | PvList | PvGuard | PvObject | PvModRef | PvTime | PvCapToken
+
+PvList = Array[PactValue]
+
+PvGuard = Guard
+
+PvObject = { f : PactValue | f in {String} } 
+// Fields can be any string, but their value is a pact value:
+
+PvModRef = { "refName":String, "refSpec":Array[ModuleName]) 
+// See # Misc type for module names
+
+PvTime = { "time" : String } | { "timep" : String }
+// There's specific formatting here depending on the UTC denom.
+// See https://github.com/kadena-io/pact/blob/e72d86749f5d65ac8d6e07a7652dd2ffb468607b/src/Pact/Types/Codec.hs#L150
+// for further details.
+
+PvCapToken = { "ctName":String, "ctArgs":Array[PactValue] } 
+// Only after pact 5:
+
+ModuleName = { "name":String, "namespace":String? } 
+// Misc type
```

Comment:
Reaching the end here, I am unsure of the utility of the JSON definitions here. I think likely the whole json specification should be contiguous, and in one spot that's clearly labeled ass `JSON representation returned by the chain` or something.

Only in the case of keysets is this useful for providing it in terms of user data (I think). Otherwise, it's primarily useful for people working with guards that are receiving chain data in return.

Reply:
I created a separate reference page for this. If these specs exist for other types of data, maybe I just need pointers to them and not to include it in doc. I'm not sure. Linda asked me about including it.